### PR TITLE
install and loads sctp kernel modules needed by AMF

### PR DESCRIPTION
**Pull request by vjayaramrh**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This PR adds code that checks to ensure the sctp kernel module (needed by AMF) is installed and loaded on the host OS.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by vjayaramrh** [comment]
FYI @johnbelamaric @s3wong @electrocucaracha @rravindran123

**Comment by electrocucaracha** [comment]
@vjayaramrh [it seems like](http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/169/e2e/1689025829377413120#1:build-log.txt%3A9785) you have to add quotes into the instruction. BTW, don't forget to include some additional unit test [here](https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/roles/bootstrap/molecule/default/tests/test_default.py)

**Comment by vjayaramrh** [comment]
> @vjayaramrh [it seems like](http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/169/e2e/1689025829377413120#1:build-log.txt%3A9785) you have to add quotes into the instruction. BTW, don't forget to include some additional unit test [here](https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/roles/bootstrap/molecule/default/tests/test_default.py)

Thanks, I missed out adding the change from the test VM to the PR, Shall push an updated PR with the fix.

**Comment by electrocucaracha** [comment]
What about adding the `kernel-modules-extra` package into the [vars file](https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/roles/bootstrap/vars/redhat.yml)? That could simplify the instructions

**Comment by vjayaramrh** [comment]
@johnbelamaric not sure why 009.sh is failing on this PR. @electrocucaracha pointed out the issue being at http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/169/e2e/1689299107576811520#1:build-log.txt%3A62816 and that the corresponding code is at https://github.com/nephio-project/test-infra/blob/main/e2e/tests/009.sh#L97 ... Would appreciate if you could review to see if its an issue in the 009.sh script? Thanks

**Comment by johnbelamaric** [comment]
Ok, yeah, the retry code is supposed to work around this issue, but the retry code itself is not working...it was hard to test due to the intermittent nature of the problem. I suppose I could have done a better job of it anyway.

In any case, since kpt is returning non-zero, the pipefail and/or errexit settings are resulting in the script being aborted. Fix is in https://github.com/nephio-project/test-infra/pull/170

If that goes through, we should merge it and then rebase this PR.

**Review by electrocucaracha — CHANGES_REQUESTED** [review]

File: e2e/provision/playbooks/roles/bootstrap/README.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ flowchart TD
     A[main.yml] -->|Check Host requirements| B(Checking host has minimum number of CPUS)
     B --> C(Checking host has minimum RAM)
     C --> D(Checking host has minimal kernel version)
-    D -->|Load gtp5g kernel module| E(Load OS specific variables)
+    D -->|Load gtp5g/sctp kernel module| E(Load OS specific variables)
```
[the remark this answers is not in the document]
Comment (vjayaramrh) [reply e2e/provision/playbooks/roles/bootstrap/README.md:98]: @electrocucaracha DO the lines L110 to L112 need to be replaced by above suggestion?

File: e2e/provision/playbooks/roles/bootstrap/README.md [hunk 2]
```diff
@@ -107,7 +107,9 @@ flowchart TD
     M -- true --> N(Configure gtp5g module)
     N --> O(Install gtp5g module)
     M -- false --> P
-    O --> P(Force all notified handlers to run at this point)
+    O -->|Load sctp kernel module| O1(install kernel-modules-extra)
+    O1 --> O2(load the sctp module)
+    O2 --> P(Force all notified handlers to run at this point)
```
Comment (electrocucaracha) [inline e2e/provision/playbooks/roles/bootstrap/README.md:112]:
```suggestion
    M -- false --> P1
    O -->|Load sctp kernel module| P1(install kernel-modules-extra)
    P1 --> P2(load the sctp module)
    P2 --> P(Force all notified handlers to run at this point)
```

File: e2e/provision/playbooks/roles/bootstrap/tasks/load-sctp-module.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Apache License, Version 2.0
+# which accompanies this distribution, and is available at
+# http://www.apache.org/licenses/LICENSE-2.0
+##############################################################################
+
+- name: install sctp on {{ ansible_os_family }}
+  become: yes
+  ansible.builtin.package:
+    name: "kernel-modules-extra-{{ ansible_kernel }}"
```
Comment (electrocucaracha) [inline e2e/provision/playbooks/roles/bootstrap/tasks/load-sctp-module.yml:14]: I'm getting the following error

```console
TASK [bootstrap : install sctp on RedHat] **************************************
fatal: [fedora34]: FAILED! => {"changed": false, "failures": ["No package kernel-modules-extra-5.17.12-100.fc34.x86_64 available."], "msg": "Failed to install some of the specified packages", "rc": 1, "results": []}
```

Can you try to use the generic package `kernel-modules-extra`?
```suggestion
    name: kernel-modules-extra
```

**Review by electrocucaracha — CHANGES_REQUESTED** [review]

File: e2e/provision/playbooks/roles/bootstrap/tasks/load-sctp-module.yml (hunk 3 above)
Comment (vjayaramrh) [reply e2e/provision/playbooks/roles/bootstrap/tasks/load-sctp-module.yml:14]: strange, It works fine on my Fedora34 cloud image gcp VM. Its important that the package being pulled matches the kernel version installed.

**Comment by vjayaramrh** [comment]
@johnbelamaric I see failure at http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/169/e2e/1689711809331728384#1:build-log.txt%3A69688 ? Is this something similar to what Sagar was seeing? I am seeing this for the first time

**Comment by johnbelamaric** [comment]
> @johnbelamaric I see failure at http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/169/e2e/1689711809331728384#1:build-log.txt%3A69688 ? Is this something similar to what Sagar was seeing? I am seeing this for the first time

It could be that, yeah. It's another flake :(

**Comment by electrocucaracha** [comment]
> @johnbelamaric I see failure at http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/169/e2e/1689711809331728384#1:build-log.txt%3A69688 ? Is this something similar to what Sagar was seeing? I am seeing this for the first time

But that time was caused by a timeout for waiting to the tunnel http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/169/e2e/1689711809331728384#1:build-log.txt%3A68817

**Pull request merged**